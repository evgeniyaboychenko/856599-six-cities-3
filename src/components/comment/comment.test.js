import React from 'react';
import renderer from 'react-test-renderer';
import Comment from './comment.jsx';

const DESCRIPTIONS = [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`];
const AVATARS = [`avatar-angelina.jpg`, `avatar-max.jpg`];
const NAMES = [`Angelina`, `Max`, `Kate`, `Pol`, `Mike`];

const comment =
{
  id: `1`,
  avatar: AVATARS[0],
  name: NAMES[0],
  rating: 2,
  text: DESCRIPTIONS[0],
  date: 1661954344,
};

it(`should Comment render correctly`, () => {
  const tree = renderer.create(
      <Comment
        comment = {comment}
        key = {comment.id}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
