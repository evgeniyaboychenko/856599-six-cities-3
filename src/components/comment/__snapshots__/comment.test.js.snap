// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Comment render correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="avatar-max.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Max
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": 39.2,
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.
    </p>
    <time
      className="reviews__time"
      dateTime="March 2020"
    >
      March 2020
    </time>
  </div>
</li>
`;
